src/readqps.jl

```diff
@@ -526,31 +622,31 @@ function readqps(filename::String)
         if section == "RANGES"
             ranges_section_read && error("more than one RANGES section specified")
             rhs_section_read || error("RHS section must come before RANGES section")
-            read_ranges_section!(qps, connames, contypes, lcon, ucon)
+            rngname = read_ranges_section!(qps, conindices, contypes, lcon, ucon)
             ranges_section_read = true
             # @show lcon
             # @show ucon
         end
         if section == "BOUNDS"
             bounds_section_read && error("more than one BOUNDS section specified")
             columns_section_read || error("COLUMNS section must come before BOUNDS section")
-            lvar, uvar = read_bounds_section(qps, varnames)
+            bndname, lvar, uvar = read_bounds_section(qps, nvar, varindices)
             bounds_section_read = true
             # @show lvar
             # @show uvar
         end
         if section == "QUADOBJ"
             columns_section_read || error("COLUMNS section must come before QUADOBJ section")
-            qrows, qcols, qvals = read_quadobj_section(qps, varnames)
+            qrows, qcols, qvals = read_quadobj_section(qps, varindices)
             quadobj_section_read = true
         end
     end
     close(qps)
 
     endata_read || @error("reached end of file before ENDATA section")
 
-    nvar = length(keys(varnames))
-    ncon = length(keys(connames))
+    nvar = length(contypes)
```

Comment:
`contypes` doesn't give the number of variables or constraints. Did you mean `conindices`? Anyways, `read_rows_section()` returned `ncon` and `read_columns_section()` returned `nvar`.

Reply:
same as above